examples/vector-layer.js

```diff
@@ -10,20 +10,9 @@ const vectorLayer = new VectorLayer({
     url: 'https://openlayers.org/data/vector/ecoregions.json',
     format: new GeoJSON(),
   }),
-  style: [
-    {
-      // if features have a COLOR property, use that in the style
-      filter: ['get', 'COLOR'],
-      style: {
-        'fill-color': ['get', 'COLOR'],
-      },
-    },
-    {
-      // otherwise, use this style
-      else: true,
-      style: {'fill-color': '#eee'},
-    },
-  ],
+  style: {
+    'fill-color': ['string', ['get', 'COLOR'], '#eee'],
+  },
```

Comment:
It was a bit of a hack to use a `filter` to evaluate the color and then have a fallback style.  The `string` type assertion with a fallback is well suited for this.